### ISSUE TITLE
CheckTypehintCallerTypeRule  and abstract base-classes

### ISSUE THREAD

**Issue by staabm** [symplify/phpstan-rules#24]
I think CheckTypehintCallerTypeRule  should not report a error when the current type is a abstract base type

```php

abstract class baseConcept {}

class impl1 extends baseConcept {}

class MyService {
    public function run(impl1 $arg)
    {
        $this->isCheck($arg);
    }
    private function isCheck(baseConcept $arg) 
    {
    }
}
```

isCheck is meant to work with any concrete subclass of `baseConcept`, therefore it should not be suggested by CheckTypehintCallerTypeRule   to use the more narrow `impl1` type.

wdyt?

**Comment by TomasVotruba** [comment]
This case is valid, as only specific type enters the `isCheck()`

**Comment by staabm** [comment]
If I understood you right, we both expect that with the above code sample the rule should not error.

I have created a PR which shows that atm a error is reported https://github.com/symplify/phpstan-rules/pull/26

**Comment by staabm** [comment]
fixed with https://github.com/symplify/phpstan-rules/pull/26

**Comment by TomasVotruba** [comment]
11.3.0 is out now :tada:
